src/explicit-constraint-set.cc

```diff
@@ -346,18 +346,12 @@ namespace hpp {
     (const size_type& i, vectorIn_t arg)
     {
       Data& d = data_[i];
-      d.qin = RowBlockIndices (d.constraint->inputConf ()).rview(arg);
-      d.constraint->explicitFunction ()->value(d.f_value, d.qin);
-      d.qout = RowBlockIndices (d.constraint->outputConf ()).rview(arg);
-      d.res_qout.vector() = d.qout;
-      vector_t rhs = d.res_qout - d.f_value;
-      d.equalityIndices.lview(d.rhs_explicit) = d.equalityIndices.rview(rhs);
-
       // compute right hand side of implicit formulation that might be
       // different (RelativePose)
       d.constraint->function ().value (d.h_value, arg);
-      rhs = d.h_value.vector ();
+      vector_t rhs = d.h_value.vector ();
       d.equalityIndices.lview(d.rhs_implicit) = d.equalityIndices.rview(rhs);
+      d.constraint->implicitToExplicitRhs(d.rhs_implicit, d.rhs_explicit);
```

Comment:
Is this preferable to computing the explicit right hand side from the input argument ?

Reply:
The comparison types must act on the implicit formulation. In the previous version, I ran into a bug where the Equality was applied on the wrong coordinate. I think we have been lucky (or unlucky) not to find this bug earlier.
In the case of a gripper grasping an object,
  - the explicit right hand side is the position of the object in the frame of its position when grasped, while
  - the implicit right hand side is the position of the handle in the gripper frame.

Reply:
The comparison types must act on the implicit formulation. In the previous version, I ran into a bug where the Equality was applied on the wrong coordinate. I think we have been lucky (or unlucky) not to find this bug earlier.
In the case of a gripper grasping an object,
  - the explicit right hand side is the position of the object in the frame of its position when grasped, while
  - the implicit right hand side is the position of the handle in the gripper frame.